flat_mutation_reader.hh

```diff
@@ -583,11 +583,9 @@ future<> consume_mutation_fragments_until(
         ConsumeEndOfStream&& consume_eos,
         db::timeout_clock::time_point timeout) {
     return do_until([stop] { return stop(); }, [&r, stop, consume_mf, consume_eos, timeout] {
-        while (!r.is_buffer_empty()) {
+        if (!r.is_buffer_empty()) {
```

Comment:
Isn't this an overkill to return a future on every fragment? Maybe we could just limit the number of loop iterations? For example:

```
auto count = std::min(32, r.buffer().size());
while (count--) {
    consume_mf(r.pop_mutation_fragment());
    if (stop()) {
        return make_ready_future<>();
    }
}
```

Reply:
The compiler should be able to see that (in do_until) the future is always ready and kill the code that checks if it's not ready or if it's exceptional.

Reply:
Actually, it can't see that, because fill_buffer() can fail and return a non-ready future.

Reply:
Maybe this

```
-           if (stop()) {
+           if (stop() || need_preempt()) {
```

Reply:
So the loop runs quickly if we don't need to preempt.

Reply:
sounds reasonable. will push new version.